core/server/common/src/test/java/alluxio/master/journal/raft/SnapshotReplicationManagerTest.java

```diff
@@ -220,6 +249,48 @@ public void requestSnapshotEqualTermHigherIndex() throws Exception {
     validateSnapshotFile(mLeaderStore, 0, 2);
   }
 
+  @Test
+  public void failGetInfoEqualTermHigherIndex() throws Exception {
+    before(2);
+    List<Follower> followers = new ArrayList<>(mFollowers.values());
+    Follower firstFollower = followers.get(0);
+    Follower secondFollower = followers.get(1);
+
+    createSnapshotFile(firstFollower.mStore); // create default 0, 1 snapshot
+    createSnapshotFile(secondFollower.mStore, 0, 2); // preferable to the default 0, 1 snapshot
+    // the second follower will not reply to the getInfo request, so the leader will request from
+    // the first after a timeout
+    secondFollower.disableGetInfo();
+
+    mLeaderSnapshotManager.maybeCopySnapshotFromFollower();
+
+    CommonUtils.waitFor("leader snapshot to complete",
+        () -> mLeaderSnapshotManager.maybeCopySnapshotFromFollower() != -1, mWaitOptions);
+    // verify that the leader still requests and get the snapshot from the first follower
+    validateSnapshotFile(mLeaderStore, 0, 1);
+  }
+
+  @Test
+  public void failSnapshotRequestEqualTermHigherIndex() throws Exception {
+    before(2);
+    List<Follower> followers = new ArrayList<>(mFollowers.values());
+    Follower firstFollower = followers.get(0);
+    Follower secondFollower = followers.get(1);
+
+    createSnapshotFile(firstFollower.mStore); // create default 0, 1 snapshot
+    createSnapshotFile(secondFollower.mStore, 0, 2); // preferable to the default 0, 1 snapshot
+    // the second follower will not start the snapshot upload, so the leader will request from the
+    // first after a timeout
+    secondFollower.disableFollowerUpload();
+
+    mLeaderSnapshotManager.maybeCopySnapshotFromFollower();
+
+    CommonUtils.waitFor("leader snapshot to complete",
+        () -> mLeaderSnapshotManager.maybeCopySnapshotFromFollower() != -1, mWaitOptions);
+    // verify that the leader still requests and get the snapshot from the first follower
+    validateSnapshotFile(mLeaderStore, 0, 1);
+  }
+
```

Comment:
I wrote this test to go alongside my other comment about the `try` block. It tests a failure from the `requestData` method.

```suggestion
@Test
  public void failFailThenSuccess() throws Exception {
    before(3);
    List<Follower> followers = new ArrayList<>(mFollowers.values());
    Follower firstFollower = followers.get(0);
    Follower secondFollower = followers.get(1);

    createSnapshotFile(firstFollower.mStore, 0, 1);
    createSnapshotFile(secondFollower.mStore, 0, 1);

    firstFollower.disableFollowerUpload();
    secondFollower.disableGetInfo();

    mLeaderSnapshotManager.maybeCopySnapshotFromFollower();

    try {
      CommonUtils.waitForResult("upload failure",
          () -> mLeaderSnapshotManager.maybeCopySnapshotFromFollower(),
          (num) -> num == 1,
          WaitForOptions.defaults().setInterval(10).setTimeoutMs(100));
    } catch (Exception e) {
      // expected to fail: no snapshot could be uploaded
    }

    Follower thirdFollower = followers.get(2);
    createSnapshotFile(thirdFollower.mStore, 0, 2);
    mLeaderSnapshotManager.maybeCopySnapshotFromFollower();
    CommonUtils.waitForResult("upload failure",
        () -> mLeaderSnapshotManager.maybeCopySnapshotFromFollower(),
        (num) -> num == 2, mWaitOptions);
    validateSnapshotFile(mLeaderStore, 0, 2);
  }
```